cpp/CMake/PythonUtils.cmake

```diff
@@ -125,13 +124,16 @@ if(ARCTICDB_FIND_PYTHON_DEV_MODE)
     if(NOT Python_EXECUTABLE AND NOT Python_ROOT_DIR AND NOT Python_LIBRARY)
         # FindPython searches the PATH environment last, but that's arguably the only correct place it should look
         find_program(Python_EXECUTABLE NAMES python3 python NAMES_PER_DIR REQUIRED NO_CMAKE_SYSTEM_PATH)
+        set(PYTHON_EXECUTABLE ${Python_EXECUTABLE} CACHE FILEPATH "Python executable found by FindPython")
```

Comment:
I don't think this needs to be cached variable.

Reply:
Looks like it needs to be or the pybind11 cmake can't resolve to the correct Python (see [this run](https://github.com/man-group/ArcticDB/actions/runs/12413039790/job/34654113561))

Reply:
Oh I didn't see that it's not part of the CMakeLists.txt files but a separate file. Probably set with [parent scope](https://cmake.org/cmake/help/latest/command/set.html#normal) could work but it's too annoying to get it right.